### PR TITLE
[PYSPARK] Fix internal python code issue

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

1. wrap the code with correct delimiter

Before:
```
{"code":"spark.sparkContext.setJobGroup(07753dd9-804e-478f-b84f-bf0735732334, ..., true)","cmd":"run_code"}
```

After:
```
{"code":"spark.sparkContext.setJobGroup('07753dd9-804e-478f-b84f-bf0735732334', '...', True)","cmd":"run_code"}
```

2. using cancelJobGroup for pyspark

Before:
```
'SparkContext' object has no attribute 'clearJobGroup'
```
After:
Using SparkContext.cancelJobGroup

3. Simplify the internal python code and throw exception on failure
We can not trust the user code is formatted correctly and we shall ensure the internal python code is simple and correct to prevent code correctness and even cause result out of sequence.
Such as, the user code might be below(maybe user invoke executePython api)
```
spark.sql('123\'\n\b\t'
```

It is difficult to escape the user code and set the job description as the statement as.
So, in this pr, I simplify the job description, just record its statementId, user can check the original code from log or on UI I think.
### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
image

**Comment by turboFei** [comment]
```
{"code":"spark.sparkContext.setJobGroup('991250a5-0cb6-42c0-b438-160133616a2d', '\nspark.sql(\"set hello=kyuubi\").show()\n', True)","cmd":"run_code"} -> {"msg_type": "execute_reply", "content": {"ename": "SyntaxError", "evalue": "EOL while scanning string literal (<stdin>, line 1)", "traceback": ["  File \"<stdin>\", line 1\n", "    spark.sparkContext.setJobGroup('991250a5-0cb6-42c0-b438-160133616a2d', '\n", "                                                                            ^\n", "SyntaxError: EOL while scanning string literal\n"], "status": "error"}}

```

**Comment by turboFei** [comment]
image

out of sequence

**Comment by turboFei** [comment]
all UT passed. @cfmcgrady

**Comment by cfmcgrady** [comment]
thank you for the fix. cc @bowenliang123 @pan3793

**Review by cfmcgrady — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala [hunk 1]
```diff
@@ -129,14 +130,20 @@ class ExecutePython(
...
     }
 
   override protected def withLocalProperties[T](f: => T): T = {
     try {
-      worker.runCode("spark.sparkContext.setJobGroup" +
-        s"($statementId, $redactedStatement, $forceCancel)")
+      // to prevent the transferred set job group python code broken
+      val jobDesc = s"Python statement: $statementId"
+      // for python, the boolean value is capitalized
+      val pythonForceCancel = forceCancel.toString.capitalize
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala:142]:
```suggestion
      val pythonForceCancel = if (forceCancel) "True" else "False"
```

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala [hunk 2]
```diff
@@ -168,15 +176,38 @@ case class SessionPythonWorker(
...
+      lock.lock()
+      block
+    } finally lock.unlock()
+  }
+
+  /**
+   * Run the python code and return the response. This method maybe invoked internally,
+   * such as setJobGroup and cancelJobGroup, if the internal python is not formatted correct,
+   * it might impact the code correctness and even cause result out of sequence. To prevent that,
+   * please make sure the internal python code simple and set internal flay, to be aware of the
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala:192]: typo? flay => flag

**Review by pan3793 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merging to master

**Comment by bowenliang123** [comment]
late LGTM, thx for the fix.

**Pull request closed without merging**